### PR TITLE
Module 5 Modules and Services

### PR DESCRIPTION
**Pull request by TashaUA**

### CODE REVIEW

File: src/app/core/services/courses.service.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  addCourse(newCourse: Course) {
+    console.log('new course was added', newCourse);
+  }
+
+  updateCourse(id: number, newCourseData: Course) {
+    console.log('update course', newCourseData);
+  }
+
+  deleteCourse(id: number) {
```
Comment (goremchik) [inline src/app/core/services/courses.service.ts:22]: Logic to open a dialog should not be in service. And it should not be a confirm method

File: src/app/core/services/courses.service.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import { Course } from "@core/models/course";
+import { CoursesData } from "@mock/courses";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CoursesService {
+
+  constructor() { }
```
Comment (goremchik) [inline src/app/core/services/courses.service.ts]: Remove

File: src/app/core/services/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+
+  constructor() { }
+
+  login(user: string) {
+    localStorage.setItem('user', JSON.stringify({ login: user, token: Math.random().toString(36).substr(2) }));
```
Comment (goremchik) [inline src/app/core/services/auth.service.ts]: I would suggest you to create a separate service for storage. You can ready here about it: https://medium.com/fafnur/localstorage-sessionstorage-%D0%B4%D0%BB%D1%8F-angular-universal-a1393940a29f
And you can do simplified version of LocalStorage.
Or you can use this package: [@ng-web-apis/common](https://www.npmjs.com/package/@ng-web-apis/common)

File: src/app/core/services/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    localStorage.removeItem('user');
+    console.log('logout is successful');
+  }
+
+  isAuthenticated() {
+    console.log('is auth!!!!!!!!', !!localStorage.getItem('user'));
+   return !!localStorage.getItem('user');
+  }
+
+  GetUserInfo() {
```
Comment (goremchik) [inline src/app/core/services/auth.service.ts]: from small letter

File: src/app/modules/components/courses/add-course/add-course.component.ts [hunk 5]
```diff
@@ -7,7 +9,10 @@ import { Component } from '@angular/core';
 })
 export class AddCourseComponent {
 
-  onAddCourse () {
-    console.log('Add new course button clicked');
+  constructor(private coursesService: CoursesService) {}
+
+  onAddCourse(): void {
+    const newCourse: Course  = { id: 5, title: 'Newly added course', creationDate: '22/05/2022' };
+    this.coursesService.addCourse(newCourse);
```
Comment (goremchik) [inline src/app/modules/components/courses/add-course/add-course.component.ts:16]: Add course component will call dialog or open new page, but will not call service

File: src/app/shared/components/header/header.component.ts [hunk 6]
```diff
@@ -7,7 +8,18 @@ import { Component, OnInit } from '@angular/core';
...
+    const user: string  = 'fakeuser';
+    this.authService.login(user);
+  }
+
+  logout(e: any) {
+    e.preventDefault();
+    this.authService.logout();
+  }
 
   ngOnInit(): void {
```
Comment (goremchik) [inline src/app/shared/components/header/header.component.ts]: Remove empty functions/constructors

File: src/app/shared/directives/auth/is-authenticated.directive.spec.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { IsAuthenticatedDirective } from './is-authenticated.directive';
+import {TemplateRef, ViewContainerRef} from "@angular/core";
+import {AuthService} from "@core/services/auth.service";
+
+describe('IsAuthenticatedDirective', () => {
+  it('should have gray border', () => {
+
```
Comment (goremchik) [inline src/app/shared/directives/auth/is-authenticated.directive.spec.ts:9]: Where a test?

File: src/app/shared/components/breadcrumbs/breadcrumbs.component.html [hunk 8]
```diff
@@ -1 +1 @@
-<p class="breadcrumbs">Courses</p>
+<p *appIsAuthenticated class="breadcrumbs">Courses</p>
```
Comment (goremchik) [inline src/app/shared/components/breadcrumbs/breadcrumbs.component.html:1]: Maybe you don't need. Such logic and this directive. show or not breadcrumbs.

Comment (TashaUA) [reply src/app/shared/components/breadcrumbs/breadcrumbs.component.html:1]: it was a part of HomeTask
Reuse header/footer components. Implement logic to hide breadcrumbs and user info block. Create custom structural directive *ifAuthenticated
Maybe I understood smth wrong

File: src/app/core/services/courses.service.ts (hunk 1 above)
Comment (TashaUA) [reply src/app/core/services/courses.service.ts:22]: moved confirm logic to component

File: src/app/core/services/auth.service.ts (hunk 3 above)
Comment (TashaUA) [reply src/app/core/services/auth.service.ts]: used 2nd point

File: src/app/modules/components/courses/add-course/add-course.component.ts (hunk 5 above)
Comment (TashaUA) [reply src/app/modules/components/courses/add-course/add-course.component.ts:16]: Will change in next tasks

**Pull request merged**